ironic_tempest_plugin/tests/scenario/ironic_standalone/test_cleaning.py

```diff
@@ -227,3 +232,123 @@ class BaremetalIdracWSManManagementCleaning(
 
     management_interface = 'idrac-wsman'
     power_interface = 'idrac-wsman'
+
+
+class BaremetalIdracRaidCleaning(bsm.BaremetalStandaloneScenarioTest):
+
+    mandatory_attr = ['driver', 'raid_interface']
+
+    image_ref = CONF.baremetal.whole_disk_image_ref
+    wholedisk_image = True
+    storage_inventory = CONF.baremetal.storage_inventory
+    driver = 'idrac'
+    api_microversion = '1.32'
+    delete_node = False
+
+    @classmethod
+    def skip_checks(cls):
+        super(BaremetalIdracRaidCleaning, cls).skip_checks()
+        if not CONF.baremetal.storage_inventory:
+            raise cls.skipException('Storage inventory file path missing.'
+                                    'Skipping Test case')
+
+    def storage_inventory_validation(self):
+        """Validates the storage information passed in file using JSON schema.
+
+        This method validates a storage inventory information against
+        a storage inventory schema.
+
+        :raises: skipException if validation of the storage inventory
+        fails and it skips Test case execution.
+
+        """
+        try:
+            with open(self.storage_inventory, 'r') as storage_invent_fobj:
+                storage_inventory_info = json.load(storage_invent_fobj)
+        except IOError:
+            raise self.skipException('Storage inventory file is not found,'
+                                     'Skipping Test Case')
+        STORAGE_INVENTORY_SCHEMA = os.path.join(os.path.dirname(
+            __file__), 'storage_inventory_schema.json')
+        with open(STORAGE_INVENTORY_SCHEMA, 'r') as storage_schema_fobj:
+            storage_inventory_schema = json.load(storage_schema_fobj)
+        try:
+            jsonschema.validate(storage_inventory_info,
+                                storage_inventory_schema)
+        except json_schema_exc.ValidationError as e:
+            msg = _("Storage Inventory validation error: %s") % e.message
+            raise self.skipException(msg)
+        return storage_inventory_info
+
+    @decorators.idempotent_id('8a908a3c-f2af-48fb-8553-9163715aa403')
+    @utils.services('image', 'network')
```

Comment:
we can add or discuss to add this as a slow test "@decorators.attr(type='slow')" as it takes around 30-40mins to run the test case.
https://docs.openstack.org/tempest/latest/HACKING.html#test-attributes : Look for Slow attribute section